<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>reverseC: This package contains methods to run mediation analysis and MR with reverse causality — reverseC-package • reverseC</title>

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



<meta property="og:title" content="reverseC: This package contains methods to run mediation analysis and MR with reverse causality — reverseC-package" />

<meta property="og:description" content="This package contains methods to run mediation analysis and MR with reverse causality. It contains functions to simulate the performance of mediation analysis and three MR methods (Egger regression, IVW and median) and functions which provide estimates and p-values of direct and indirect effects given a dataset." />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">reverseC</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/SharonLutz/reverseC">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>reverseC: This package contains methods to run mediation analysis and MR with reverse causality</h1>
    <small class="dont-index">Source: <a href='https://github.com/SharonLutz/reverseC/blob/master/R/reverseC.R'><code>R/reverseC.R</code></a></small>
    <div class="hidden name"><code>reverseC-package.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>This package contains methods to run mediation analysis and MR with reverse causality. It contains functions to simulate the performance of mediation analysis and three MR methods (Egger regression, IVW and median) and functions which provide estimates and p-values of direct and indirect effects given a dataset.</p>
    
    </div>

        
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>This package examines the performance of Mendelian Randomization (MR) and Mediation Analysis methods in the presence of reverse causality.</p>
    
    <h2 class="hasAnchor" id="reversec-for-mendelian-randomization"><a class="anchor" href="#reversec-for-mendelian-randomization"></a>reverseC for Mendelian Randomization</h2>

    
    <p>The reverseMRsim function in the reverseC R package examines the performance of Mendelian Randomization (MR) methods in the presence of reverse causality. Through simulation studies, this R function examines the type 1 error rate and power for 3 popular MR methods when the role of the intermediate phenotype and outcome were correctly specified and when they were reversed (i.e. reverse causality).</p>
    
    <h2 class="hasAnchor" id="mendelian-randomization-input"><a class="anchor" href="#mendelian-randomization-input"></a>Mendelian Randomization Input</h2>

    
    <p>nSNP is the number of SNPs generated from a binomial distribution for n subjects (input n) for a given minor allele frequency (input vector MAF).</p>
<p>For the SNPs Xi, the mediator/ exposure M is generated from a normal distribution with the variance (input varM) and the mean as follows:</p>
<p>\(E[M] = \gamma_0 + \Sigma \gamma_x * X_i\)</p>
<p>All of these values are inputted by the user (i.e. the intercept gamma0, and the genetic effect size as a vector gammaX).</p>
<p>The outcome Y is generated from a normal distribution with the variance (input varY) and the mean as follows:</p> 
<p>\(E[Y] = \beta_0 + \beta_M * M\)</p>
<p>All of these values are inputted by the user (i.e. the intercept beta0 and the effect of the mediator directly on the outcome as betaM).</p>
<p>After the SNPs X, mediator M, and outcome Y are generated, then the reverseMRsim function compares the power and type 1 error rate of the following 3 methods to detect the path from M to Y: Egger Regression, the Median Weighted Approach, and the Inverse Variance Weighted (IVW) Approach.</p>
    
    <h2 class="hasAnchor" id="mendelian-randomization-example"><a class="anchor" href="#mendelian-randomization-example"></a>Mendelian Randomization Example</h2>

    
    <p>For 1,000 subjects (n=1000), we generated 10 SNPs (nSNP=10) with a minor allele frequency of 20% (specified by MAF) that have a genetic effect size of 0.4 (specified by gammaX) on the normally distributed mediator and the mediator has an effect size varying from 0, 0.2 to 0.3 (specified by betaM) on the normally distributed outcome. We considered 3 MR approaches: Egger Regression, the Median Weighted Approach, and the Inverse Variance Weighted (IVW) Approach.</p>
    
    <h2 class="hasAnchor" id="mendelian-randomization-output"><a class="anchor" href="#mendelian-randomization-output"></a>Mendelian Randomization Output</h2>

    
    <p>For the example, we get corresponding plot. In the plot below, the methods ending in NR have the true outcome as the outcome where as the methods ending in R have the true outcome reversed with the mediator. When the mediator and outcome are reversed, the Egger regression and the Median Weighted Approach have an inflated type 1 error rate. While the IVW approach does not have an inflated type 1 error rate, there is very little difference in the IVW approach if the mediator and outcome are reversed, which implies that this approach cannot easily distinguish the causal relationship between the mediator and outcome.</p>
<p><img src='figures/reverseMRplot.png' alt='alt="MR_PLOT_image_placeholder"' /></p>
    
    <h2 class="hasAnchor" id="reversec-for-mediation-analysis"><a class="anchor" href="#reversec-for-mediation-analysis"></a>reverseC for Mediation Analysis</h2>

    
    <p>These functions examines the performance of mediation analysis methods in the presence of reverse causality.</p>
    
    <h2 class="hasAnchor" id="mediation-analysis-example"><a class="anchor" href="#mediation-analysis-example"></a>Mediation Analysis Example</h2>

    
    <pre>
library(reverseC)
?reverseMAsim # For details on this function
    reverseMAsim(n = 1000, pX = 0.2, gamma0 = 0, gammaX = 0.2, varM = 1, beta0 = 0, betaX = 0, 
             betaM = c(0.1, 0.2, 0.3), varY = 1, nSim = 500, nSimImai = 500, SEED = 1, plot.pdf = T, 
             plot.name = "reverseMAplot.pdf", alpha_level = 0.05)
    reverseMAsim(n = 1000, pX = 0.2, gamma0 = 0, gammaX = 0, varM = 1, beta0 = 0, betaX = 0.2, 
             betaM = c(0.1, 0.2, 0.3), varY = 1, nSim = 500, nSimImai = 500, SEED = 1, plot.pdf = T, 
             plot.name = "reverseMAplotDirect.pdf", alpha_level = 0.05)
    reverseMAsim(n = 1000, pX = 0.2, gamma0 = 0, gammaX = 0.2, varM = 1, beta0 = 0, betaX = 0.2, 
             betaM = c(0.1, 0.2, 0.3), varY = 1, nSim = 500, nSimImai = 500, SEED = 1, plot.pdf = T, 
             plot.name = "reverseMAplotBoth.pdf", alpha_level = 0.05)
</pre>
    
    <h2 class="hasAnchor" id="mediation-analysis-output"><a class="anchor" href="#mediation-analysis-output"></a>Mediation Analysis Output</h2>

    
    <p>The example code produces this plot:
<img src='figures/reverseMAplot.png' alt='alt="MA_PLOT_image_placeholder"' /></p>
    
    <h2 class="hasAnchor" id="installation"><a class="anchor" href="#installation"></a>Installation</h2>

    
    <pre>
install.packages("devtools") # devtools must be installed first
devtools::install_github("SharonLutz/reverseC")
</pre>
    
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>MR.Egger is the Egger Regression approach to MR.</p><pre>
Bowden J., Davey Smith G., &amp; Burgess S. (2015). Mendelian Randomization 
with invalid instruments: effect estimation and bias detection through 
Egger regression. International Journal of Epidemiology, 44(2), 512-525. 
</pre><p>MR.IVW is the Inverse Variant Weighted approach to MR.</p><pre>
Burgess, S., Butterworth, A., &amp; Thompson, S. G. (2013). Mendelian 
Randomization Analysis With Multiple Genetic Variants Using Summarized 
Data. Genetic Epidemiology, 37(7), 658-665.
</pre><p>MR. Median is the Median Weighted approach to MR.</p><pre>
Bowden, J., Davey Smith, G., Haycock, P. C., &amp; Burgess, S. (2016). Consistent 
Estimation in Mendelian Randomization with Some Invalid Instruments Using a 
Weighted Median Estimator. Genetic Epidemiology, 40(4), 304-314. 
</pre>
    
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p>Useful links:</p><ul>
<li><p><a href='https://github.com/SharonLutz/reverseC'>https://github.com/SharonLutz/reverseC</a></p></li>
<li><p>Report bugs at <a href='https://github.com/SharonLutz/reverseC/issues'>https://github.com/SharonLutz/reverseC/issues</a></p></li>
</ul></div>
    

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      
      <li><a href="#details">Details</a></li>

      <li><a href="#reversec-for-mendelian-randomization">reverseC for Mendelian Randomization</a></li>

      <li><a href="#mendelian-randomization-input">Mendelian Randomization Input</a></li>

      <li><a href="#mendelian-randomization-example">Mendelian Randomization Example</a></li>

      <li><a href="#mendelian-randomization-output">Mendelian Randomization Output</a></li>

      <li><a href="#reversec-for-mediation-analysis">reverseC for Mediation Analysis</a></li>

      <li><a href="#mediation-analysis-example">Mediation Analysis Example</a></li>

      <li><a href="#mediation-analysis-output">Mediation Analysis Output</a></li>

      <li><a href="#installation">Installation</a></li>

      <li><a href="#references">References</a></li>

      <li><a href="#see-also">See also</a></li>
          </ul>

    <h2>Author</h2>
    <p><strong>Maintainer</strong>: Annie Thwing <a href='mailto:annie.thwing@ucdenver.edu'>annie.thwing@ucdenver.edu</a></p>
<p>Authors:</p><ul>
<li><p>Sharon Lutz</p></li>
<li><p>Michael Gooch <a href='mailto:goochmi@gmail.com'>goochmi@gmail.com</a> [contributor]</p></li>
</ul>


  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Annie Thwing, Sharon Lutz, Michael Gooch.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
   </div>

  

  </body>
</html>

